stackslib/src/net/connection.rs

```diff
@@ -364,6 +372,9 @@ pub struct ConnectionOptions {
     /// how long a mempool sync is allowed to take, in total, before timing out
     pub mempool_sync_timeout: u64,
 
+    /// a channel for submitting long-running tasks to the stacks-node
+    pub async_rpc_channel: Option<SyncSender<AsyncRequests>>,
```

Comment:
Hmm, I don't think this is the right place for this.  I think this belongs instead in `ConversationHttp`, which is the struct concerned with managing the lifecycle of an HTTP connection.  `ConnectionOptions` is not meant to contain any information that gets initialized at runtime.

Reply:
Seeing that you're setting up the `Receiver` in `neon_node.rs`, maybe the initial `Sender` should go into the `HttpPeer` instance within `PeerNetwork`.  Then, it can give any `ConversationHttp` instances a copy whenever a new client is accepted.

Reply:
@kantai added this part, maybe he can explain

Reply:
Adding it as an argument to `HttpPeer::new` is fine with me -- I added it to `ConnectionOptions` because my understanding was that struct was meant to encapsulate all of the configuration arguments to network peer structs, etc to avoid cluttering those interfaces. I will update.

Reply:
I moved it to `HttpPeer::new()` in latest commit. Let me know if that looks right to you.